cmd/openshift-install/create.go

```diff
@@ -623,3 +677,83 @@ The cluster should be accessible for troubleshooting as detailed in the document
 https://docs.openshift.com/container-platform/latest/support/troubleshooting/troubleshooting-installations.html
 The 'wait-for install-complete' subcommand can then be used to continue the installation`)
 }
+
+func getClusterOperatorNames(ctx context.Context, cc *configclient.Clientset) ([]string, error) {
+	listCtx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	cos, err := cc.ConfigV1().ClusterOperators().List(listCtx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	names := []string{}
+	for _, v := range cos.Items {
+		names = append(names, v.Name)
+	}
+	return names, nil
+}
+
+type coStabilityStatus struct {
+	name, reason, message string
+	err                   error
+}
+
+func getCOProgressingStatus(ctx context.Context, cc *configclient.Clientset, name string) coStabilityStatus {
+	stabilityTimeout := 5 * time.Minute
+	stabilityContext, cancel := context.WithTimeout(context.Background(), stabilityTimeout)
+	defer cancel()
+
+	//fields.OneTermEqualSelector("objectMeta.name", name))
```

Comment:
without this line, doesn't the code below return true after the first clusteroperator has been stable?

Reply:
you are correct, sir! I couldn't figure out the value should be, turns out metadata.name is the answer.